sensor/common/collectorruntimeconfig/handler.go

```diff
@@ -0,0 +1,125 @@
+package collectorruntimeconfig
+
+import (
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/sensor/common"
+	"github.com/stackrox/rox/sensor/common/message"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// Store is a store for network graph external sources.
+//
+//go:generate mockgen-wrapper
+type Store interface {
+	CollectorConfigValueStream() concurrency.ReadOnlyValueStream[*sensor.CollectorConfig]
+}
+
+// Handler forwards the collector runtime config received from Central to Collectors.
+type Handler interface {
+	common.SensorComponent
+}
+
+type handlerImpl struct {
+	stopSig   concurrency.Signal
+	updateSig concurrency.Signal
+
+	collectorConfig            *sensor.CollectorConfig
+	collectorConfigProtoStream *concurrency.ValueStream[*sensor.CollectorConfig]
+
+	lock sync.Mutex
+}
+
+func (h *handlerImpl) Start() error {
+	go h.run()
+	return nil
+}
+
+func (h *handlerImpl) Stop(_ error) {
+	h.stopSig.Signal()
+}
+
+func (h *handlerImpl) Notify(common.SensorComponentEvent) {}
+
+func (h *handlerImpl) Capabilities() []centralsensor.SensorCapability {
+	// return []centralsensor.SensorCapability{centralsensor.NetworkGraphExternalSrcsCap}
+	return nil
+}
+
+func getCollectorConfig(msg *central.MsgToSensor) *sensor.CollectorConfig {
+	if clusterConfig := msg.GetClusterConfig(); clusterConfig != nil {
+		log.Info("Created fake collector config")
```

Comment:
I will remove all of these logging statements before merging.

Reply:
Thanks for calling this out. To further ensure reviewers are not distracted by temporary logging or testing you should mark statements that you intend to remove with a clear comment like `// TODO: Remove before merging`

Reply:
Done